### DISCUSSION TITLE
std::format - what exactly happens at compile-time?

### DISCUSSION THREAD

**Discussion by nuuSolutions** [microsoft/STL#4088] [Q&A] [answered]

**Comment by jovibor** [comment, marked as answer]
Compilation.

**Comment by nuuSolutions** [comment]
some very smart people here. Thanks!. Are you a Microsoft MVP?

**Comment by StephanTLavavej** [comment]
A less hilarious but more relevant answer:

`<format>` performs compile-time *checking* of format strings, but doesn't do any other work at compile-time that could somehow accelerate run-time processing. The checking is driven by the behavior of `consteval`, which (unlike plain `constexpr`) forces compile-time evaluation even when the top-level call happens at runtime. For part of the involved logic, see here:

https://github.com/microsoft/STL/blob/1c59a205787f4fdcd9c20decc8fa6dfa9096d394/stl/inc/format#L3490-L3495
